### PR TITLE
Add support for changing the volume with the mouse wheel

### PR DESCRIPTION
**Pull request by xoascf**
Change the volume by rolling the mouse wheel over the volume icon in the notification area.

_Maybe add it as a toggleable option, or change some things earlier by adding or using another ManagedShell function._

### CODE REVIEW

**Comment by dremin** [comment]
I'm good with this being in RetroBar for now rather than ManagedShell since it sends the command to Explorer. We can move it to ManagedShell if/when the audio API is used for this.

**Comment by xoascf** [comment]
> I'm good with this being in RetroBar for now rather than ManagedShell since it sends the command to Explorer. We can move it to ManagedShell if/when the audio API is used for this.

I was thinking of adding the list of missing APPCOMMAND constants and making public the default tray icon constants in ManagedShell, so they can be used here without having to define them again.

File: RetroBar/Controls/NotifyIcon.xaml.cs [hunk 1]
```diff
@@ -111,5 +111,33 @@ private void NotifyIcon_OnMouseMove(object sender, MouseEventArgs e)
...
+
+        private const int APPCOMMAND_VOLUME_DOWN = 0x90000;
+        private const int APPCOMMAND_VOLUME_UP = 0xA0000;
+        private const string VOLUME_GUID = "7820ae73-23e3-4229-82c1-e41cb67d5b9c";
+
+        private void ChangeVolume(bool upOrDown)
+        {
+            NativeMethods.SendMessage(WindowHelper.FindWindowsTray(IntPtr.Zero), (int)NativeMethods.WM.APPCOMMAND,
+                WindowHelper.FindWindowsTray(IntPtr.Zero), upOrDown ? (IntPtr)APPCOMMAND_VOLUME_UP : (IntPtr)APPCOMMAND_VOLUME_DOWN);
+        }
```
Comment (dremin) [inline RetroBar/Controls/NotifyIcon.xaml.cs:123]: Could you move this functionality out from the UserControl into a utility class?

**Review by dremin — CHANGES_REQUESTED** [review]
I like the idea of moving the constants to ManagedShell but this is good for now, once the volume code is moved out of the UI code into a utility/helper class.

**Review by dremin — APPROVED** [review]
🎉

**Comment by dremin** [comment]
> Could we add the option to adjust the volume from anywhere in the system tray, rather than just over the volume icon? It's such a tiny target to aim for, where the system tray is easy to 'hit'.

That could break compatibility--any tray icon can potentially respond to the mouse wheel, and it wouldn't be great to break that functionality.

**Comment by xoascf** [comment]
> Could we add the option to adjust the volume from anywhere in the system tray, rather than just over the volume icon? It's such a tiny target to aim for, where the system tray is easy to 'hit'.

There is a freeware software called Volume² that can help you with that: https://github.com/irzyxa/Volume2/releases

**Pull request merged**